aea/mail/base.py

```diff
@@ -334,9 +378,43 @@ def __init__(
         )
         if isinstance(message, Message):
             message = self._check_consistency(message, to, sender)
+
         self._to = to
         self._sender = sender
-        self._protocol_id = protocol_id
+
+        # set protocol_id and protocol_specification_id
+        if isinstance(message, Message):
```

Comment:
In this case, enforce that no `protocol_id` and `specification_id` are provided.

Reply:
it was done in other parts of code, but refactored